src/nvim/os/process.c

```diff
@@ -71,24 +72,30 @@ static bool os_proc_tree_kill_rec(HANDLE process, int sig)
   return (bool)TerminateProcess(process, (unsigned int)sig);
 }
 /// Kills process `pid` and its descendants recursively.
-bool os_proc_tree_kill(int pid, int sig)
+bool os_proc_tree_kill(Process *proc, int sig)
 {
   assert(sig >= 0);
   assert(sig == SIGTERM || sig == SIGKILL);
   if (pid > 0) {
     ILOG("terminating process tree: %d", pid);
     HANDLE h = OpenProcess(PROCESS_ALL_ACCESS, false, (DWORD)pid);
-    return os_proc_tree_kill_rec(h, sig);
+    bool rv = os_proc_tree_kill_rec(h, sig);
+    if (rv) {
+      // uv_process_kill sets this manually (for Windows), but is not used by
+      // use (and on Unix uv_kill is used directly).
+      proc->exit_signal = sig;
+    }
```

Comment:
Would rather change the return type that require Process parameter.

Reply:
I don't understand.  Do you mean to set exit_signal in the caller?

Reply:
Given the last commit (5a6b169): there should be a function wrapping this then probably?!

Reply:
OTOH `os_proc_tree_kill` is proc specific by its name, and it should have a new function when providing this for a `kill()` user function?

Reply:
? os_proc is the prefix for this module.  Not every function here needs a Process parameter, and an extra function isn't needed.